site/content/coursework2.md

```diff
@@ -0,0 +1,424 @@
+```
+
+## Part 1: Reimplementing the beam
+
+In the previous coursework you implemented a 5-point stencil for the Euler-Bernoulli beam
+We will solve the stationary variant of this equation (i.e. no time-component this time).
+In particular, we will solve a three-point bending test. The beam is supported at both ends,
+and at the centre a concentrated load is applied.
+
+$$
+\delta(x-\frac{1}{2}) = -k \partial_{xxxx}u(x)
+$$
+with a small material parameter $k$. Here $\delta$ is a Dirac function, it is one at the centre
+of the beam $x=\frac{1}{2}$ and zero everywhere else.
+
+Boundary conditions remain as before:
+u(0,t)=u(1,t)=0
+$$
+and
+$$
+\partial_{xx}u(0,t) = \partial_{xx}u(1,t) = 0.
+$$
+
+
+To implement this create a class `Bernoulli5pt` that discretises the
+spatial operator using a 2nd order accurate 5-point stencil. Watch out
+for the boundary conditions.
+
+{{< hint info >}}
+Hint: Have a look at `tests/test_one_dim.py`
+{{< /hint >}}
+
+{{< hint info >}}
+You should do your implementation for this part in a file called
+`part1_euler_bernoulli.py` placed in the root of the repository.
+{{< /hint >}}
+
+
+The analytical solution is
+$$w(x) = \begin{cases}
+    \frac{x(4x^2-3)}{48k}, & \mbox{for } 0 < x < \frac{1}{2} \\
+    \frac{(x-1)(1-8x+4x^2)}{48k}, & \mbox{for } \frac{1}{2} < x < 1
+    \end{cases}$$
+but some constant is wrong in the above.
+
+{{< question "Part 1a questions" >}}
+1. How does the error in your numerical solution behave under grid
+   refinement? Can you explain what you see?
+{{< /question >}}
+
+
+
+
+## Part 2: The 3D Darcy Equation
+
+
+Discretise the equation
+$$
+\partial_t u - \nabla \cdot K(x, y, z) \nabla u = f(x, y, z)
+$$
+on the cubic domain $\Omega = [0, 1] \times [0, 1] \times [0, 1]$,
+using forward Euler as a timestepping scheme.
+
+To do this, create a class `Poisson7pt` that discretises the spatial
+operator using a 2nd order accurate 7-point stencil (as derived in lectures).
+$$
+-\nabla \cdot K(x, y, z) \nabla u.
+$$
+
+Using
+$$
+K(x, y, z) = cos(\pi y)
+$$
+$$
+f(x, y, z) =  (\pi+3\pi^2) \sin(\pi x)\sin(\pi y)\sin(\pi z),
+$$
+Dirichlet boundary conditions on the boundary
+$$
+u(x, y, z) = 0.
+$$
+The exact solution for this
+problem is
+$$
+u^*(x, y, z) = \sin(\pi x)\sin(\pi y)\sin(\pi z).
+$$
+
+
+### Part 2a: timestepping
+{{< hint info >}}
+You should do your implementation for this part in a file called
+`part2_explicit_euler.py` placed in the root of the repository.
+{{< /hint >}}
+
+Implement an explicit Euler time-stepping scheme using your 7-point stencil.
+
+{{< hint info >}}
+Ensure that your implementation works correctly in parallel as well as
+serial (when run with MPI).
+{{< /hint >}}
+
+{{< question "Part 2a questions" >}}
+1. How does the error in your numerical solution behave under grid
+   refinement? Can you explain what you see?
+2. What restriction, if any, is there on the size of the timestep you can
+   choose?
+{{< /question >}}

```

Comment:
We've kind of covered timestepping in the 1D coursework, do we want to go round again here, or go straight to stationary state?

Reply:
I agree perhaps the question is a duplicate. I ask them to compare to the stationary equation, at least that will be a new question